### ISSUE TITLE
Rename debug option

### ISSUE THREAD

**Issue by tjconcept** [substack/node-browserify#1061]
This is my subjective opinion and I might as well be wrong, but I really don't like option names like "debug".

- it says little/nothing about actual effect
- it forces you to lookup the option in the documentation
- it can be very subjective - I might not like the same things during debug as someone else

Apparently the only effect in browserify is to enable source maps - then why not name it "sourceMaps" or similar? I keep thinking it might be doing something else, or that some transforms use it for something that they didn't document properly. It feels to implicit.

What do you think?

**Comment by Fishrock123** [comment]
:+1:

**Comment by mynameistechno** [comment]
I agree. I was hesitant to enable this for non-dev envs as I wasn't sure what else `--debug` might do.

**Comment by goto-bus-stop** [comment]
yes, `--source-map` or `--source-maps` or something would be a good name. We could alias it to `debug` in the minimist config in https://github.com/browserify/browserify/blob/cbdf69e256f2cc4fc8025dadb01551ff5be54b3c/bin/args.js.

Realistically we need to keep the old name around for a very long time because all the plugins and transforms that support source maps check for the `.debug` property on browserify's options object, but we could move to a new name in docs.
